packages/bridge/src/vl/VLBridge.ts

```diff
@@ -457,17 +465,28 @@ export default class VLBridge {
   private _rebuildFilters() {
     let newFilter = this._combineFilters(
       this._vizFilters
-      .filter((hasFilter) => hasFilter.filter !== null)
-      .map((hasFilter) => hasFilter.filter)
+        .filter((hasFilter) => hasFilter.filter !== null)
+        .map((hasFilter) => hasFilter.filter)
     );
 
     // Update (if required) the readonly layer
     if (this._readOnlyLayer) {
       this._readOnlyLayer.viz.filter.blendTo(newFilter, 0);
     }
 
-    if (this._animation && this._animation.propertyName === 'filter') {
-      newFilter = `@${this._animation.variableName} and ${newFilter}`;
+    if (this._layer.viz.filter.isAnimated()) {
+      const filterExpr = this._layer.viz.filter;
+      const animationFilter = VL_BINARY_EXPRESSION_TYPES.indexOf(filterExpr.expressionName) > -1
+        ? filterExpr.a.expressionName === 'animation' ? filterExpr.a : filterExpr.b
+        : filterExpr;
+
+      // TODO change animation.toString()
```

Comment:
Note: CARTO VL issue https://github.com/CartoDB/carto-vl/issues/1381

Reply:
It would be nice if you can add the link of the issue in the TODO